File: src/packages/utils/src/workspace.ts
/**
 * Copyright (c) 2023-present Plane Software, Inc. and contributors
 * SPDX-License-Identifier: AGPL-3.0-only
 * See the LICENSE file for details.
 */

// plane imports
import type { IWorkspace } from "@plane/types";

export const orderWorkspacesList = (workspaces: IWorkspace[]): IWorkspace[] =>
  workspaces.sort((a, b) => a.name.localeCompare(b.name));

File: apps/api/plane/api/urls/workspace.py
# Copyright (c) 2023-present Plane Software, Inc. and contributors
# SPDX-License-Identifier: AGPL-3.0-only
# See the LICENSE file for details.

from django.urls import path

from plane.api.views import (
    WorkspaceListCreateAPIEndpoint,
    WorkspaceDetailAPIEndpoint,
)

urlpatterns = [
    path(
        "workspaces/",
        WorkspaceListCreateAPIEndpoint.as_view(http_method_names=["get", "post"]),
        name="workspaces",
    ),
    path(
        "workspaces/<str:slug>/",
        WorkspaceDetailAPIEndpoint.as_view(http_method_names=["get", "patch"]),
        name="workspace",
    ),
]
